File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLSendMessageGeoLocationAction.java
package com.github.badoualy.telegram.tl.api;

import static com.github.badoualy.telegram.tl.StreamUtils.*;

import java.lang.Override;
import java.lang.String;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLSendMessageGeoLocationAction extends TLAbsSendMessageAction {
    public static final int CLASS_ID = 0x176f8ba1;

    public TLSendMessageGeoLocationAction() {
    }

    @Override
    public String toString() {
        return "sendMessageGeoLocationAction#176f8ba1";
    }

    @Override
    public int getClassId() {
        return CLASS_ID;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLSendMessageGeoLocationAction.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String

/**
 * sendMessageGeoLocationAction#176f8ba1
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLSendMessageGeoLocationAction : TLAbsSendMessageAction() {
    private val _constructor: String = "sendMessageGeoLocationAction#176f8ba1"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLSendMessageGeoLocationAction) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x176f8ba1.toInt()
    }
}
